optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -727,6 +727,32 @@ public ScoreDefinition getScoreDefinition() {
     // Model methods
     // ************************************************************************
 
+    public EntityDescriptor deduceEntityDescriptor(Class<?> entityClass) {
+        EntityDescriptor<Solution_> entityDescriptor;
+        if (entityClass != null) {
+            entityDescriptor = getEntityDescriptorStrict(entityClass);
+            if (entityDescriptor == null) {
+                throw new IllegalArgumentException("The config (" + this
+                        + ") has an entityClass (" + entityClass + ") that is not a known planning entity.\n"
+                        + "Check your solver configuration. If that class (" + entityClass.getSimpleName()
+                        + ") is not in the entityClassSet (" + getEntityClassSet()
+                        + "), check your " + PlanningSolution.class.getSimpleName()
+                        + " implementation's annotated methods too.");
+            }
+        } else {
+            Collection<EntityDescriptor<Solution_>> entityDescriptors = getGenuineEntityDescriptors();
+            if (entityDescriptors.size() != 1) {
+                throw new IllegalArgumentException("The config (" + this
+                        + ") has no entityClass (" + entityClass
+                        + ") configured and because there are multiple in the entityClassSet ("
+                        + getEntityClassSet()
+                        + "), it cannot be deduced automatically.");
+            }
+            entityDescriptor = entityDescriptors.iterator().next();
+        }
+        return entityDescriptor;
+    }
+
```

Comment:
Same comment on the `entityClass` nullity. This is two methods masquerading as one.

Reply:
Regarding the last three comments - I agree, my plan was to look into it in the next PR, which focuses on the remaining part of the config code - including config of `ValueSelector` and `EntitySelector`. For this PR, I just moved these existing methods to - in my opinion - a more suitable place without changing their internals.

I should have clearly stated my intentions before opening the PR for review.